### PR TITLE
Fix shebang even if first line doesn't start with '#!'

### LINKED ISSUE #3179: Allow injecting shebang (now it only replaces - eg. touching files starting with '#!')
The shebang regex is here:
https://github.com/easybuilders/easybuild-framework/blob/develop/easybuild/framework/easyblock.py#L2189

### PR DESCRIPTION
**Pull request by Darkless012**
fixes #3179

### CODE REVIEW

File: easybuild/framework/easyblock.py [hunk 1]
```diff
@@ -2209,6 +2209,9 @@ def fix_shebang(self):
                         if should_patch:
                             contents = shebang_regex.sub(shebang, contents)
                             write_file(path, contents)
+                        else:
```
Comment (boegel) [inline easybuild/framework/easyblock.py]: @Darkless012 This is not fully correct...

Imagine I have a bash script called `bin/foo` that has `#!/bin/bash`, and `fix_python_shebang_for = ['bin/*']` is used.

You don't want to inject `#!/usr/bin/env python` in that case and break the script.

We should make sure there's *no* shebang at all before adding it. Maybe like this:

```python
elif not contents.startswith('#!/'):

File: test/framework/toy_build.py [hunk 2]
```diff
@@ -2390,6 +2392,8 @@ def test_fix_shebang(self):
             "   'echo \"#!/usr/bin/perl -w\\n# test\" > %(installdir)s/bin/t4.pl',",
             # space after #! + 'env perl5'
             "   'echo \"#!/usr/bin/env perl5\\n# test\" > %(installdir)s/bin/t5.pl',",
+            # no shebang perl
+            "   'echo \"# test\" > %(installdir)s/bin/t6.pl',",
```
Comment (boegel) [inline test/framework/toy_build.py]: Also add a test which has `#!/bin/bash` as shebang, to make sure you don't break scripts that do have a shebang?

File: easybuild/framework/easyblock.py (hunk 1 above)
Comment (akesandgren) [reply easybuild/framework/easyblock.py]: Does the code path also verify that it is a script in the first place?? And not a random binary?

Comment (Darkless012) [reply easybuild/framework/easyblock.py]: @boegel Yup, I've also thought about having wrong files "fixed". Also, there is optional space possible after `#!` I will add that to tests as well.

Comment (Darkless012) [reply easybuild/framework/easyblock.py]: @akesandgren Yup, the files are read and checked for shebang regex or `#!` string. So binary files will not be touched. https://github.com/easybuilders/easybuild-framework/blob/1d112fb3aac5c62b58e80dc5bd15b2ebb1188513/easybuild/framework/easyblock.py#L2200-L2205

File: test/framework/toy_build.py (hunk 2 above)
Comment (Darkless012) [reply test/framework/toy_build.py]: Done

File: easybuild/framework/easyblock.py (hunk 1 above)
Comment (boegel) [reply easybuild/framework/easyblock.py]: @akesandgren Confirmed, that was fixed in #3029

**Pull request merged**